DfE.FindInformationAcademiesTrusts/Pages/Trusts/TrustsAreaModel.cs

```diff
@@ -7,19 +7,42 @@ namespace DfE.FindInformationAcademiesTrusts.Pages.Trusts;
 public class TrustsAreaModel : PageModel, ITrustsAreaModel
 {
     private readonly ITrustProvider _trustProvider;
+    protected readonly IDataSourceProvider DataSourceProvider;
+    private readonly ILogger<TrustsAreaModel> _logger;
 
-    public TrustsAreaModel(ITrustProvider trustProvider, string pageName)
+    public TrustsAreaModel(ITrustProvider trustProvider, IDataSourceProvider dataSourceProvider,
+        ILogger<TrustsAreaModel> logger, string pageName)
     {
         _trustProvider = trustProvider;
+        DataSourceProvider = dataSourceProvider;
+        _logger = logger;
         PageName = pageName;
     }
 
     [BindProperty(SupportsGet = true)] public string Uid { get; set; } = "";
     public Trust Trust { get; set; } = default!;
+    public List<DataSourceListEntry> DataSources { get; set; } = new();
     public string PageName { get; init; }
     public string? PageTitle { get; init; }
     public string Section => ViewConstants.AboutTheTrustSectionName;
 
+    public string MapDataSourceToName(DataSource source)
+    {
+        switch (source.Source)
+        {
+            case Source.Gias:
+            case Source.Mstr:
+                return "Get information about schools";
+            case Source.Cdm:
+                return "RSD Service Support team";
+            case Source.Mis:
+                return "State-funded school inspections and outcomes: management information";
+            default:
+                _logger.LogError("Data source does not map to known type");
```

Comment:
It would be good to include the datasource object in the log (using structured logging)